### DISCUSSION TITLE
PAGView可以动态设置path吗

### DISCUSSION THREAD

**Discussion by YauLam** [Tencent/libpag#2341] [Q&A] [answered]
2个问题，如下：
1.一个PAGView对应两个path，每次点击，根据判断条件，选择set哪一个path。官方建议这么操作吗，这么操作会有什么问题吗？
2.flush的使用方式，有个是否执行动画的开关，执行调用pagView.play，play里面执行了flush；如果不执行动画，setProgress=1.0，这是需要手动调用flush吗。setProgress里面调用了一个 native update的方法，这个native update做了些什么呢。目前我尝试setProgress后，没有调用flush，也能正常直行到我想要的结果。

**Comment by kevingpqi123** [comment, marked as answer]
1、PAGView 可以通过 setPath 的方式切换 pag 文件，但不建议在 play 状态下切换，pag 内部是线程安全的，播放状态下切换可能会存在锁等待，低端机上有可能会存在卡顿问题，先播放暂停然后再切换；如果是 2~ 3 个 pag 文件固定切换，由于 pag 文件播放的首帧涉及到环境的初始化和缓存的构建，相对会比较耗时，推荐的做法是使用 pag 的组合模式，将 2 ~ 3 个 pag 文件放在同一个自己创建的 PAGComposition 中，通过 PAGLayer 的 setVisible 方法控制隐藏与显示，这种效率是最高的，也不用考虑播放停止后再切换

2、在 Android 平台由于历史原因，部分版本 setProgress 的时候默认执行了 flush，在 API 涉及层面 setProgress 和 flush 是两个独立的操作，我们推荐 setProgress 后执行 flush 方法，setProgress 仅仅是控制进度，后续我们会移除 update 的操作， 即便执行了两次 flush，这里对于性能也没有任何影响，pag 内部会检测如果渲染内容没有发生变化是不会触发再次绘制的

**Reply by YauLam** [reply]
感谢回答

**Comment by peter100u** [comment]
推荐的做法是使用 pag 的组合模式，将 2 ~ 3 个 pag 文件放在同一个自己创建的 PAGComposition 中，通过 PAGLayer 的 setVisible 方法控制隐藏与显示，这种效率是最高的，也不用考虑播放停止后再切换,
这段有参考代码吗
